System/Process/Common.hs

```diff
@@ -0,0 +1,215 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE RecordWildCards #-}
+module System.Process.Common
+    ( CreateProcess (..)
+    , CmdSpec (..)
+    , StdStream (..)
+    , ProcessHandle(..)
+    , ProcessHandle__(..)
+    , withFilePathException
+    , PHANDLE
+    , modifyProcessHandle
+    , withProcessHandle
+    , fd_stdin
+    , fd_stdout
+    , fd_stderr
+    , mbFd
+    , mbPipe
+    , pfdToHandle
+
+-- Avoid a warning on Windows
+#ifdef WINDOWS
+    , CGid (..)
```

Comment:
I think we're on the same page: I'd rather _not_ export the `CGid` data constructor. That's what you're saying too, right? In fact, that's what this PR does for the public-facing API. The .Common module is not exported from the package, and the .Internals module does not export the `CGid` data constructor.